src/app/content/search/hooks/index.ts

```diff
@@ -97,11 +97,14 @@ export const receiveSearchHook: ActionHookBody<typeof receiveSearchResults> = (s
 };
 
 export const clearSearchHook: ActionHookBody<typeof clearSearch | typeof openToc> = (services) => () => {
-  const scrollTarget = selectNavigation.scrollTarget(services.getState());
-  if (scrollTarget && isSearchScrollTarget(scrollTarget)) {
+  const state = services.getState();
+  const query = selectNavigation.persistentQueryParameters(state).query;
+  const systemQueryParams = selectNavigation.systemQueryParameters(state);
+
+  if (query) {
     services.history.replace({
       hash: '',
-      search: '',
+      search: queryString.stringify(systemQueryParams),
     });
   }
```

Comment:
@TomWoodward this fixes the bug that was initially [reported](https://github.com/openstax/unified/issues/1631) but I'm worried that I'm breaking something else. The only thing I can think of is that if you had a target set to a highlight and then made a search and closed the search results, that highlight query would be lost. That seems ok to me, though.

Reply:
i think really you want to take whatever the current params are and remove the search query, and remove the target only if it is for a search result, but otherwise leave everything else

Reply:
updated and did the same for the hash